core/src/main/java/org/pentaho/di/core/fileinput/FileInputList.java

```diff
@@ -313,7 +316,18 @@ public static FileInputList createFolderList( VariableSpace space, String[] fold
           FileObject[] fileObjects = directoryFileObject.findFiles( new AllFileSelector() {
             @Override
             public boolean traverseDescendents( FileSelectInfo info ) {
-              return info.getDepth() == 0 || subdirs;
+              return ( info.getDepth() == 0 || subdirs )
+                // Check if one has permission to list this folder
+                && hasAccess( info.getFile() );
+            }
+
+            private boolean hasAccess( FileObject fileObject ) {
+              try {
+                return Files.isReadable( Paths.get( ( new File( fileObject.getName().getPath() ) ).toURI() ) );
```

Comment:
I think fileObject.getPublicURIString() will produce the exact same result as Paths.get( ( new File( fileObject.getName().getPath() ) ).toURI() and we would avoid creating a new File object and do the Paths.get.
Can you confirm that?

Reply:
Unfortunately, there's a bug on the Apache Commons VFS and a 'java.lang.IllegalArgumentException' exception is thrown when the path contains a space. As so, it's safer obtain the URI this way.